flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainersConfig.java

```diff
@@ -0,0 +1,599 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.tests.util.flink.container;
+
+import org.apache.flink.configuration.CheckpointingOptions;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.HighAvailabilityOptions;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.configuration.RestOptions;
+import org.apache.flink.configuration.TaskManagerOptions;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.apache.flink.configuration.HeartbeatManagerOptions.HEARTBEAT_INTERVAL;
+import static org.apache.flink.configuration.HeartbeatManagerOptions.HEARTBEAT_TIMEOUT;
+import static org.apache.flink.configuration.JobManagerOptions.SLOT_REQUEST_TIMEOUT;
+import static org.apache.flink.configuration.MetricOptions.METRIC_FETCHER_UPDATE_INTERVAL;
+
+/** The central configuration holder for Flink container-based test environments. */
+public class FlinkContainersConfig {
```

Comment:
@PatrickRen thanks for the review.
I was also a bit undecided about what to do with the `FlinkContainersBuilder`. One factor is that I wanted to keep `FlinkContainerConfig` independent of any Testcontainer-specific classes (`Network`, `GenericContainer`) - the idea is to have it purely as a generalized configuration for Flink containers and not tie it into any particular testing framework implementation.  I removed `setBaseImage` as another generic parameter, but am not convinced that  moving the rest of them to `FlinkContainerConfig` is the right move. 

Another question, if we decide to get rid of `FlinkContainersBuilder`, is where to place all of it's methods. `FlinkContainers` is mainly concerned with the lifecycle management of a the containers and interaction with them (job submission, cleanup etc.) and it is already a sizeable class. From the perspective of separation of concerns it makes sense to me to have a delegate that is responsible purely for the preparational steps that clearly preceded the responsibilities of `FlinkContainers`. Maybe `*Builder` is a confusing name, because it mixes both the builder pattern and the fact that we mean to actually _build_ the containers with Docker.

Reply:
Hi @PatrickRen, I dropped `FlinkContainerBuilder` and introduced `TestcontainersSettings`, as you proposed.
I would prefer to keep the builder patterns for the following reasons:
1) It is one of the most widespread design patterns in general and we also use them everywhere in Flink. I do not think they are confusing for the users.
2) I believe if anything, they reduce the potential confusion for the users. A user might change the settings later and expect the changes for taking effect on the execution, but it won't because we already populated concrete underlying classes with it.  Immutability makes things more clear. 
3) Inline declaration: FlinkContainers are mostly initialized as class fields and using the setter style would require every test implementation to either use static{} initialization blocks or to provide initialization methods similar to what is done with the [Flink config ](https://github.com/apache/flink/blob/89f782416bf5b600ce35e3e3e0edd4992019f2e6/flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SmokeKafkaITCase.java#L99). I find it pretty convenient to be able to configure those test fields inline.

Looking forward to your feedback!